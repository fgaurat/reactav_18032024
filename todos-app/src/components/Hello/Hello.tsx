import { useState } from "react"

function Hello() {

    const[count,setCount]=useState(0)
  return (
    <>
    HelloWorld
    <br />
    <p>
    count : {count}
    </p>
    
    <button onClick={() => setCount(c =>c+1)}>inc</button>

    </>
  )
}

export default Hello