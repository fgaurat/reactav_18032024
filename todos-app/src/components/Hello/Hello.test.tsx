import { describe, expect, test } from "vitest";
import { render,screen } from "@testing-library/react";
import Hello from "./Hello";
import userEvent from "@testing-library/user-event"


describe('Test Hello',()=>{

    test("render Hello component",()=>{
        render(<Hello/>)
    })

    test("Snapshot render Hello component",()=>{
        const result = render(<Hello/>)
        expect(result).toMatchSnapshot()
    })


    test("HelloWorld string in Hello component",()=>{
        render(<Hello/>)
        const s = screen.getByText(/HelloWorld/)
        expect(s).toBeInTheDocument()

    })

    test("count:0 in Hello component",()=>{
        render(<Hello/>)
        // const s = screen.getByText(/count : 0/)
        const s = screen.getByText("count : 0")
        expect(s).toBeInTheDocument()

    })
    test("increment count in Hello component",async ()=>{
        
        //Arrange
        render(<Hello/>)
        
        //Act
        const button = screen.getByText(/inc/)
        await userEvent.click(button)
        
        // Assert
        await screen.findByText("count : 1")



    })


})