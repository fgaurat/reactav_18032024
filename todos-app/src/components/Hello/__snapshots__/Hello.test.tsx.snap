// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test Hello > Snapshot render Hello component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      HelloWorld
      <br />
      <p>
        count : 
        0
      </p>
      <button>
        inc
      </button>
    </div>
  </body>,
  "container": <div>
    HelloWorld
    <br />
    <p>
      count : 
      0
    </p>
    <button>
      inc
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
